drivers/pinctrl/pinctrl_stm32.c

```diff
@@ -0,0 +1,574 @@
+/*
+ * Copyright (c) 2018 Bobby Noelte
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/** @file
+ * @brief STM32 pinctrl implementation
+ * @defgroup device_driver_pinctrl_stm32 STM32 pinctrl inplementation
+ *
+ * A common driver for STM32 pinctrl. SoC specific adaptions are
+ * done by device tree and soc.h.
+ *
+ * @{
+ */
+
+/** Compile if at least one device is activated
+ * @code{.codegen}
+ * compatible = 'st,stm32-pinctrl'
+ * codegen.out_include('templates/dts/basic_tmpl.c')
+ * if_device_tree_controller_compatible('PIN', compatible)
+ * @endcode{.codegen}
+ */
+/** @code{.codeins}@endcode */
+
+#include <autoconf.h>
+#include <generated_dts_board.h>
+#include <soc.h>
+
+#include <pinctrl_common.h>
+#include <dt-bindings/clock/stm32_clock.h>
+#include <dt-bindings/pinctrl/pinctrl_stm32.h>
+
+#include <errno.h>
+#include <string.h>
+#include <zephyr/types.h>
+#include <misc/__assert.h>
+#include <misc/util.h>
+#include <kernel.h>
+#include <device.h>
+#include <gpio.h>
+#include <pinctrl.h>
+
+#include <clock_control/stm32_clock_control.h>
+
+/**
+ * @def PINCTRL_STM32_GPIO_COUNT
+ * @brief Number of GPIO ports that may be managed by the pinctrl driver.
+ *
+ * This is the total number of GPIO ports available. The number of aktivated
+ * GPIO ports may be less.
+ */
+/* GPIOx is defined by the STM32 CubeMX library. */
+#if defined(GPIOK)
+#define PINCTRL_STM32_GPIO_COUNT 11
+#elif defined(GPIOJ)
+#define PINCTRL_STM32_GPIO_COUNT 10
+#elif defined(GPIOI)
+#define PINCTRL_STM32_GPIO_COUNT 9
+#elif defined(GPIOH)
+#define PINCTRL_STM32_GPIO_COUNT 8
+#elif defined(GPIOG)
+#define PINCTRL_STM32_GPIO_COUNT 7
+#elif defined(GPIOF)
+#define PINCTRL_STM32_GPIO_COUNT 6
+#elif defined(GPIOE)
+#define PINCTRL_STM32_GPIO_COUNT 5
+#elif defined(GPIOD)
+#define PINCTRL_STM32_GPIO_COUNT 4
+#elif defined(GPIOC)
+#define PINCTRL_STM32_GPIO_COUNT 3
+#elif defined(GPIOB)
+#define PINCTRL_STM32_GPIO_COUNT 2
+#elif defined(GPIOA)
+#define PINCTRL_STM32_GPIO_COUNT 1
+#else
+#error "Missing GPIOx definition!"
+#endif
+
+/**
+ * @brief Extract mux value from device tree pinmux.
+ *
+ * @note Device tree pinmux expands to a tuple (pin, mux).
+ */
+#define PINCTRL_STM32_PINCONF_PINMUX_MUX(pin, mux) mux
+
+/**
+ * @brief Extract pin value from device tree pinmux.
+ *
+ * @note Device tree pinmux expands to a tuple (pin, mux).
+ */
+#define PINCTRL_STM32_PINCONF_PINMUX_PIN(pin, mux) pin
```

Comment:
These two macros appear to have been deprecated by

`def pinmux_pin(pinmux):`

should they be removed?

Reply:
Good spot. Thank you.Willr emove them.